components/hab/src/command/pkg/download.rs

```diff
@@ -50,20 +51,45 @@ use crate::{api_client::{self,
                     ChannelIdent,
                     Error as CoreError}};
 
+use habitat_common::ui::{Glyph,
+                         Status,
+                         UIWriter};
+
 use reqwest::StatusCode;
 use retry::{delay,
             retry};
 
 use crate::error::{Error,
                    Result};
 
-use habitat_common::ui::{Glyph,
-                         Status,
-                         UIWriter};
-
 pub const RETRIES: usize = 5;
 pub const RETRY_WAIT: Duration = Duration::from_millis(3000);
 
+#[derive(Debug, Deserialize)]
+pub struct PackageSetFile {
+    pub format_version:  Option<u8>,
+    pub file_descriptor: Option<String>,
```

Comment:
Not sure what the purpose of the file_descriptor field is as we are not checking it?

Reply:
It was suggested as part of making the flow of package inclusion traceable when generating the manifest. We've not started that work, but I left the optional field in to save a version rev when we do it.

Reply:
Ok that seems reasonable as long as we're planning to do that work and the design is worked out - if it's just a suggestion that may or may not come to fruition in current form I'd lean toward taking it out for now.